network/src/application/netperf/mod.rs

```diff
@@ -0,0 +1,335 @@
+                                    /* maybe add dedicated counters? but network_application_{out/in}bound_traffic
+                                     * seems to have us coverred
+                                     * */
+                                }
+
+                            }
+
+                        }
+                        _ => {/* Currently ignore all*/}
+                    }
+                }
+            }
+        }
+        warn!(
+            NetworkSchema::new(&self.network_context),
+            "{} NetPerf event listener terminated", self.network_context
+        );
+    }
+}
+#[allow(dead_code)]
+#[derive(Clone)]
+enum NetPerfCommands {
+    Broadcast,
+}
+
+fn preferred_axum_port(netperf_port: u16) -> u16 {
+    if netperf_port != 9107 {
+        let _ = OpenOptions::new()
+            .write(true)
+            .create_new(true)
+            .open("/tmp/9107.tmp");
+
+        let _ = OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(format!("/tmp/{}.tmp", netperf_port));
+    }
+    netperf_port
+}
+
+async fn start_axum(state: NetPerfState, mut netperf_port: u16) {
+    loop {
+        let app = Router::new()
+            .route("/", get(usage_handler))
+            .route("/peers", get(get_peers).layer(Extension(state.clone())))
+            .route(
+                "/command",
+                post(parse_query).layer(Extension(state.clone())),
+            );
+
+        let addr = std::net::SocketAddr::from(([0, 0, 0, 0], netperf_port));
+
+        // run it with hyper on netperf_port
+        let res = axum::Server::try_bind(&addr);
+        if let Ok(handler) = res {
+            let _ = handler.serve(app.into_make_service()).await;
+            break;
+        } else {
+            netperf_port += 1;
+        }
+    }
+    preferred_axum_port(netperf_port);
+}
+
+async fn usage_handler() -> &'static str {
+    "Usage: curl 127.0.0.01:9107/peers"
+}
+
+#[derive(Serialize)]
+struct PeerList {
+    peers: Vec<PeerId>,
+}
+
+impl PeerList {
+    pub fn new(len: usize) -> Self {
+        PeerList {
+            peers: Vec::with_capacity(len),
+        }
+    }
+}
+
+async fn get_peers(Extension(state): Extension<NetPerfState>) -> Json<PeerList> {
+    let mut out = PeerList::new(state.peer_list.len());
+
+    let connected = state.peer_list.iter();
+
+    for peer in connected {
+        out.peers.push(peer.key().to_owned());
+    }
+
+    Json(out)
+}
+
+async fn parse_query(
+    Extension(state): Extension<NetPerfState>,
+    Query(_params): Query<std::collections::HashMap<String, String>>,
+) -> impl IntoResponse {
+    //TODO(AlexM): Extract size and duration from _params
+    let size = NETPERF_DEFAULT_MSG_SIZE;
+    let duration = Duration::from_secs(NETPERF_DEFAULT_DURTAION_SEC);
+
+    spawn_named!(
+        "[NetPerf] Brodcast Task",
+        netperf_broadcast(state, size, duration)
+    );
+
+    StatusCode::OK
+}
+
+#[allow(dead_code)]
```

Comment:
Maybe add a comment that we're coming back to this? In my mind RPC is the much more interesting case 😄